### PR TITLE
File API Part I

### PR DESCRIPTION
**Pull request by ninaandrup**
#2040

Other PR: #2431

### CODE REVIEW

File: main/test/ca/uwaterloo/flix/library/TestFile.flix [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    @test
+    def testIsReadable02(): Bool & Impure =
+        isReadable("./fakeREADME.md") == Ok(false)
+
+    /////////////////////////////////////////////////////////////////////////////
+    // isSymbolicLink                                                          //
+    /////////////////////////////////////////////////////////////////////////////
+
+    // TODO: find symbolic link to test
```
Comment (ninaandrup) [inline main/test/ca/uwaterloo/flix/library/TestFile.flix:74]: I could not find a file to test for symbolic link

**Comment by ninaandrup** [comment]
I don't know if I should rewrite the test to:
https://github.com/flix/flix/pull/2431#issuecomment-967070568

File: main/test/ca/uwaterloo/flix/library/TestFile.flix (hunk 1 above)
Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/library/TestFile.flix:74]: I think for many of them one test is fine.

Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/library/TestFile.flix:74]: Just test what you can and leave the rest. We don't need to test both outcomes of each function. One outcome is enough.

**Comment by magnus-madsen** [comment]
Let me know when ready for review/merge.

**Comment by ninaandrup** [comment]
> Let me know when ready for review/merge.

It should be ready when the check passes :-)

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**